balancer/base/balancer.go

```diff
@@ -60,7 +61,7 @@ type baseBalancer struct {
 	csEvltr *balancer.ConnectivityStateEvaluator
 	state   connectivity.State
 
-	subConns map[resolver.Address]balancer.SubConn
+	subConns map[resolver.Address]AddrInfo
```

Comment:
With changes in this file, we will still unnecessarily reconnect when the weight for an address changes.
To avoid that, we should clear metadata field when addresses are received.
But this changes the behavior.

Base is to be used as an example, and in the case only picking algorithm is different.
I think we should leave `base` package as is, and re-implement this for wrr.

wrr will have some special behaviors, some I can think of now:
 - Duplicates in addresses list (same address string, but multiple entries in addrs list, with potential different metadata)
   - Should we add up the weights?
 - Since metadata is not hashed, the only useful thing is the address string itself. Maybe we should use the string as the key, instead of the address struct

Reply:
In this care diff becomes really small, just weightedroundrobin package declaration with metadata population in edsbalancer. Updated pull request.